src/pineko/check.py

```diff
@@ -161,40 +165,36 @@ def contains_ren(grid, max_as, max_al):
              max as order
     max_al : int
              max al order
+    sv_type : Scale
+        kind of scale_variation to be checked (either REN or FACT)
+
     Returns
     -------
-    bool
-        is ren scale-variation available for as
-    bool
-        is ren scale-variation available for al
+    : AvailableAtMax
+        result of the check
+    : int
+        effective max_as in the grid
     """
-    order_array = np.array([order.as_tuple() for order in grid.orders()])
-    order_mask = pineappl.grid.Order.create_mask(grid.orders(), max_as, max_al)
-    order_list = order_array[order_mask]
-    as_orders = []
-    al_orders = []
-    for order in order_list:
-        as_orders.append(order[0])
-        al_orders.append(order[1])
-    # ren sv starts one order after the first order with as
-    min_as = 1 if min(as_orders) == 0 else min(as_orders)
-    # ren sv starts one order after the first order with al
-    min_al = 1 if min(al_orders) == 0 else min(al_orders)
-    order_as_is_present = False
-    order_al_is_present = False
-    sv_as_present = False
-    sv_al_present = False
-    for order in order_list:
-        if order[0] == min_as + 1:
-            order_as_is_present = True
-            if order[-2] != 0:
-                sv_as_present = True
-        if order[1] == min_al + 1:
-            order_al_is_present = True
-            if order[-2] != 0:
-                sv_al_present = True
-    if not order_as_is_present:
-        sv_as_present = True
-    if not order_al_is_present:
-        sv_al_present = True
-    return sv_as_present, sv_al_present
+    index_to_check = sv_type.value.index
+    order_list = get_order_list(grid, max_as, max_al)
+    as_orders = pure_qcd_orders(order_list)
+    max_as_effective = max(ord[0] for ord in as_orders)
+    min_as = min(ord[0] for ord in as_orders)
+    max_as_effective_cen = max(ord[0] for ord in as_orders if ord[index_to_check] == 0)
+    max_as_effective_sv = max(
+        (ord[0] for ord in as_orders if ord[index_to_check] != 0), default=0
+    )
+    if max_as_effective_cen == max_as_effective:
+        if max_as_effective_sv == max_as_effective:
+            checkres = AvailableAtMax.BOTH
+        elif max_as_effective - min_as == 0:
```

Comment:
```suggestion
        elif max_as_effective == min_as:
```
Stupid performance tip: one operation less :)

Reply:
Fine for me but I believe the other way is more clear